### PR TITLE
Enhancement to support HCL and JSON configuration file formats

### PR DESCRIPTION
**Pull request by jacec**
Pull request adding support for HCL and JSON configuration as discussed in #127

Note: HCL and JSON configuration is converted to YAML and existing parsing code is therefore unchanged. See [HCL](https://github.com/hashicorp/hcl) for more information.

hclconfig.go added 
hclconfig_test.go added
Readme updated to reflect configuration language options.
minor modification to walter that now assumes files with a suffix of ".hcl" or ".json" are config files in those formats. ALL other suffixes are assumed to be YAML and thus this shouldn't break any existing usage.
Added HCL and JSON test files

build and all tests complete successfully.

### CODE REVIEW

**Comment by jacec** [comment]
also referenced in #18

**Comment by takahi-i** [comment]
Hello @jacec Thank you very much for the contribution!

We think HCL is a good format and consider to support the HCL format in the past (#18) but we think that supporting two format is too much for Walter team because of our resource. Therefore at this point we will stick with supporting only YAML for a while (at least major version update)...

Deeply sorry that we are not able to merge this PR...

**Comment by jacec** [comment]
That's the reason I didn't change the parsing code and simply converted the HCL/JSON []map[string]interface{} to the YAML []map[interface{}]interface to minimize the changes and support you needed. That being said, I understand your hesitation, but would urge you to consider this enhancement in the next major release as it opens up a number of benefits around pipeline configuration. Thanks!

File: README.md [hunk 1]
```diff
@@ -76,6 +78,50 @@ pipeline:
...
+JSON
+```json
+{
+  "pipeline":{
+    "stage" :{
+      "name":"command_stage_1",
+      "type":"command",
+      "command":"echo 'hello, world'"
+    },
+    "stage":{
```
Comment (ZeissS) [inline README.md]: I don't think this is valid json, as the key is used three times.

Comment (jacec) [inline README.md]: Good point, interestingly the HCL lib accepts it and parses correctly, but in real terms javascript would parse this and assume the state of the last key. Maybe we could change the top level object to ```stages``` and have an array under that. I can change it, but what are your thoughts?

Comment (ZeissS) [inline README.md]: Sorry, I haven't played around enough with HCL or Walter to give you a good recommendation here. Just noticed the JSON is invalid.

**Comment by takahi-i** [comment]
I will close this issue since the pr is not be merged currently following the previous discussion of this thread.

**Pull request closed without merging**